vrpn_Tracker_IMU.C

```diff
@@ -0,0 +1,578 @@
+/// @file vrpn_Tracker_IMU.C
+/// @copyright 2015 ReliaSolve.com
+/// @author ReliaSolve.com russ@reliasolve.com
+/// @license Standard VRPN license.
+
+#include "vrpn_Tracker_IMU.h"
+#include <cmath>
+
+#undef VERBOSE
+
+vrpn_IMU_Magnetometer::vrpn_IMU_Magnetometer
+          (std::string name, vrpn_Connection *output_con,
+            vrpn_IMU_Axis_Params params,
+           float update_rate, bool report_changes) :
+	vrpn_Analog_Server(name.c_str(), output_con),
+	d_update_interval (update_rate ? (1/update_rate) : 1.0),
+	d_report_changes (report_changes)
+{
+  // We have 3 output channels.
+  num_channel = 3;
+
+	//--------------------------------------------------------------------
+	// Set the current timestamp to "now" and current direction to along X.
+	// This is done in case we never hear from the analog devices.
+  vrpn_gettimeofday(&d_prevtime, NULL);
+  vrpn_Analog::timestamp = d_prevtime;
+  channel[0] = 1;
+  channel[1] = 0;
+  channel[2] = 0;
+
+  // Set empty axis values.
+	d_vector.params = params;
+  d_vector.time = d_prevtime;
+
+	//--------------------------------------------------------------------
+	// Open analog remote to use to listen to.
+	//   If the name starts with the "*" character, use our output
+  // connection rather than getting a new connection for it.
+	//   Set up callbacks to handle updates to the analog reports.
+	setup_vector(&d_vector);
+
+	//--------------------------------------------------------------------
+  // Set the minimum and maximum values past each other and past what
+  // we expect from any real device, so they will be inialized properly
+  // the first time we get a report.
+  for (size_t i = 0; i < 3; i++) {
+    d_mins[i] = 1e100;
+    d_maxes[i] = -1e100;
+  }
+}
+
+vrpn_IMU_Magnetometer::~vrpn_IMU_Magnetometer()
+{
+	// Tear down the analog update callbacks and remotes
+	teardown_vector(&d_vector);
+}
+
+// This routine handles updates of the analog values. The value coming in is
+// adjusted per the parameters in the axis description, and then used to
+// update the value there. The value is used by the device code in
+// mainloop() to update its internal state; that work is not done here.
+
+void	VRPN_CALLBACK vrpn_IMU_Magnetometer::handle_analog_update
+                     (void *userdata, const vrpn_ANALOGCB info)
+{
+  vrpn_IMU_Vector	*vector = (vrpn_IMU_Vector *)userdata;
```

Comment:
Yeah, I must have copied that from older code and not paid attention.  I always do static_cast now.  Will fix.